lib/rspec/matchers/built_in/respond_to.rb

```diff
@@ -125,9 +125,20 @@ def matches_arity?(actual, name)
 
           return true if expectation.empty?
 
-          signature = Support::MethodSignature.new(Support.method_handle_for(actual, name))
 
-          Support::StrictSignatureVerifier.new(signature).with_expectation(expectation).valid?
+          Support::StrictSignatureVerifier.new(
+            build_method_signature(actual, name)
+          ).with_expectation(expectation).valid?
+        end
+
+        def build_method_signature(actual, name)
```

Comment:
nit: this method doesn't do much "building" -- that's what `Support::MethodSignature` does.  Maybe call it `method_signature_for(actual, name)`?

Reply:
Ha I think I had that first then changed it, happy to oblige.